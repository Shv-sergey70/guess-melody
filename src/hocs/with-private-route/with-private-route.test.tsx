import * as React from 'react';
import {configure, shallow} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import withPrivateRoute from "./with-private-route";

configure({adapter: new Adapter()});

const Component: React.FunctionComponent = () => <div/>;

describe(`WithPrivateRoute correctly works`, () => {
  test(`Redirect to auth screen renders`, () => {
    const ComponentWrapped = withPrivateRoute(Component);

    const componentWrapped = shallow(
        <ComponentWrapped
          user={{}}
        />
    );

    expect(componentWrapped).toMatchSnapshot();
  });

  test(`Component renders`, () => {
    const ComponentWrapped = withPrivateRoute(Component);

    const componentWrapped = shallow(
        <ComponentWrapped
          user={{
            email: `test@test.ru`
          }}
        />
    );

    expect(componentWrapped).toMatchSnapshot();
  });
});
