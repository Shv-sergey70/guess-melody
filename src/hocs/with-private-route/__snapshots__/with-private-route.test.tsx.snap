// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithPrivateRoute correctly works Component renders 1`] = `ShallowWrapper {}`;

exports[`WithPrivateRoute correctly works Redirect to auth screen renders 1`] = `ShallowWrapper {}`;
